src/platform/webusb/device.rs

```diff
@@ -0,0 +1,193 @@
+use std::{sync::Arc, time::Duration};
+
+use wasm_bindgen_futures::{js_sys::Array, wasm_bindgen::JsCast, JsFuture};
+use web_sys::{js_sys::Uint8Array, UsbControlTransferParameters, UsbDevice, UsbInTransferResult};
+
+use crate::{
+    descriptors::{validate_config_descriptor, DESCRIPTOR_TYPE_CONFIGURATION},
+    transfer::{Control, EndpointType, TransferError, TransferHandle},
+    DeviceInfo, Error,
+};
+
+#[derive(Clone)]
+pub(crate) struct WebusbDevice {
+    pub device: UsbDevice,
+    config_descriptors: Vec<Vec<u8>>,
+}
+
+/// SAFETY: This is NOT safe at all.
+unsafe impl Sync for WebusbDevice {}
```

Comment:
I am not sure how to get rid of this.

In the example of probe-rs I need to use `async_trait(?Send)` which does somehow still require those bounds because the JS objects are essentially a *mut and marked as such via PhantomData.